ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/ansible_eos_device.py

```diff
@@ -84,6 +84,9 @@ def __init__(self, name: str, connection: ConnectionBase, tags: list | None = No
 
         super().__init__(name, tags, disable_cache=False)
         self.check_mode = check_mode
+        # Check the ansible connection is defined
+        if not self.check_mode and not hasattr(connection, "_sub_plugin"):
+            raise AristaAvdError(message="Unable to determine ansible connection. Please manually configure ansible_connection as `httpapi` for this host.")
```

Comment:
May be we can make it like:

```suggestion
        # Check the ansible connection is defined
        try:
            if self.check_mode or (plugin_name := connection._sub_plugin.get("name")) == ANSIBLE_EOS_PLUGIN_NAME:
                self._connection = connection
            else:
                raise AristaAvdError(message=f"The provided Ansible connection does not use EOS HttpApi plugin: {plugin_name}")
        except AttributeError:
            raise AristaAvdError(message="AVD could not determine ansible connection. Please configure ansible_connection as `httpapi` for this host.")
```

Reply:
updated